core/services/ocr/delegate.go

```diff
@@ -216,19 +217,21 @@ func (d Delegate) ServicesForSpec(jb job.Job) (services []job.ServiceCtx, err er
 			return nil, errors.New("TransmitterAddress is missing")
 		}
 
+		gasLimit := chain.Config().EvmGasLimitDefault()
+		if jb.GasLimitGwei != nil {
+			gasLimit = assets.GWei(int64(*jb.GasLimitGwei)).Uint64()
```

Comment:
I know this is a WIP PR, but I figured I'd give some early feedback.

Unfortunately this does not make sense - gas limits are not denominated in wei, gwei, or any ether denomination, they're their own unit. Gas units are just gas, not ether, and mixing up gas _limits_ with gas _prices_ is erroneous and we should avoid mixing these up in our code as it is confusing and will lead to bugs and programming (and configuration) errors.

This code also does not do what we expect it to do. For example, if I give a gas limit of, say, 100,000, this line will end up multiplying 100,000 by a billion, which will give me a gas limit of `1e14`, which exceeds the block limit by about 7 orders of magnitude. And even if we did want to specify gas limits in "gwei", as in, in 10^9 denominations, it wouldn't be very useful, because most chains have block gas limits in the 10's of millions, so we'd have to specify something like "0.0001" in the job spec to get 100,000 gas, after multiply by 10^9.

My recommendation would be to simply rename this field to `gasLimit` instead, and drop the `GWei` suffix since it's misleading, and that way job specs that specify e.g `gasLimit = 100000` actually get a gas limit of 100,000, which is the expected behavior of all API's that take gas limits as configurable parameters.

Now, if this PR is adding a gas _price_ configuration, then this would make sense, and I suggest that we call this field `gasPriceGwei` instead of `gasLimitGwei`, as gas prices and gas limits are quite different things. However, I don't think that's what you're trying to do here, given that you set `gasLimit = chain.Config().EvmGasLimitDefault()` which is a gas _limit_ setting and is not denominated in gwei or wei, or any other denomination - it's just flat gas units (1 gas = 1 gas).

Reply:
Thank you @makramkd! I will address your feedback.

Reply:
@makramkd I forgot to mention in the description: https://app.shortcut.com/chainlinklabs/story/20502/allow-gas-limit-to-be-tuned-per-job-type-and-per-job-gas-savings
we are just adding an ability to override `ETH_GAS_LIMIT_DEFAULT` on per-job basis, hence it should be the limit as you said, not a price.